### PR TITLE
[Bugfix] Fix bug drop/recover db maintain task

### LINKED ISSUE #9995: [sync materialized view]task not delete when drop database
### Steps to reproduce the behavior (Required)
```
CREATE DATABASE db1;
USE db1;
CREATE TABLE `t1` (
                                            `k1` date,
                                            `k2` int,
                                            `k3` int
                                        ) ENGINE=OLAP
                                        DUPLICATE KEY(`k1`)
                                        COMMENT "OLAP"
                                        DISTRIBUTED BY HASH(`k1`) BUCKETS 3;
CREATE MATERIALIZED VIEW mv1
                             DISTRIBUTED BY HASH(k1) BUCKETS 10
                             REFRESH ASYNC
                             AS SELECT k1,k2 from t1;
DROP DATABASE db1;
select * from information_schema.tasks WHERE `DATABASE`='db1';
```

### Expected behavior (Required)
`Empty set (0.00 sec)`
### Real behavior (Required)
```
+-----------+---------------------+-----------------+----------+----------------------------------------------------------------------------------------------+-------------+
| TASK_NAME | CREATE_TIME         | SCHEDULE        | DATABASE | DEFINITION                                                                                   | EXPIRE_TIME |
+-----------+---------------------+-----------------+----------+----------------------------------------------------------------------------------------------+-------------+
| mv-227617 | 2022-08-15 14:59:25 | EVENT_TRIGGERED | db1      | insert overwrite mv1 SELECT `db1`.`t1`.`k1` AS `k1`, `db1`.`t1`.`k2` AS `k2` FROM `db1`.`t1` | NULL        |
+-----------+---------------------+-----------------+----------+----------------------------------------------------------------------------------------------+-------------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.4-RELEASE 9744f36 |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9995

## Problem Summary(Required) ：

When the db is dropped, the entire task of the db is deleted, and when the db recovers, all tasks are reconstructed, but because the materialized view does not save the attribute information, this repair is not perfect. It may also be necessary to mention the process of PR perfect repair and reconstruction.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 56 / 76 (73.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|2|00.00%|[561, 564]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|3|00.00%|[348, 350, 353]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|12|25|48.00%|[320, 529, 530, 531, 532, 533, 534, 535, 536, 538, 3389, 3390, 3391]|
|:large_blue_circle:|com/starrocks/alter/Alter.java|2|3|66.67%|[270]|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|40|41|97.56%|[79]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java [hunk 1]
```diff
@@ -44,6 +52,66 @@ public static Task buildMvTask(MaterializedView materializedView, String dbName)
...
+                    TaskSchedule taskSchedule = new TaskSchedule(startTime, step.getLongValue(),
+                            TimeUtils.convertUnitIdentifierToTimeUnit(intervalLiteral.getUnitIdentifier().getDescription()));
+                    task.setSchedule(taskSchedule);
+                    task.setType(Constants.TaskType.PERIODICAL);
+                }
+            }
+        }
+    }
+
+    public static void mappingTaskTypeAndSetRefreshInfo(Task task, MaterializedView materializedView)
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java:92]: this function can be removed and replaced by the former one.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -3276,35 +3294,16 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
         LOG.info("Successfully create materialized view[{};{}]", mvName, mvId);
 
         // NOTE: The materialized view has been added to the database, and the following procedure cannot throw exception.
-        if (createMvSuccess) {
-            createTaskForMaterializedView(dbName, materializedView, optHints);
-        }
+        createTaskForMaterializedView(dbName, materializedView, optHints);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3297]: why remove the check of createMvSuccess?

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java [hunk 3]
```diff
@@ -44,6 +52,66 @@ public static Task buildMvTask(MaterializedView materializedView, String dbName)
...
+        previousTaskProperties.put(PartitionBasedMaterializedViewRefreshProcessor.MV_ID,
+                String.valueOf(materializedView.getId()));
+        task.setProperties(previousTaskProperties);
+        task.setDefinition(
+                "insert overwrite " + materializedView.getName() + " " + materializedView.getViewDefineSql());
+        task.setExpireTime(0L);
+        return task;
+    }
+
+    public static void mappingTaskTypeAndSetRefreshInfo(Task task, RefreshSchemeDesc refreshSchemeDesc)
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java:69]:
```suggestion
    public static void updateTaskInfo(Task task, RefreshSchemeDesc refreshSchemeDesc)
```

**Review by ABingHuang — CHANGES_REQUESTED** [review]
better to add some UT

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java (hunk 1 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/TaskBuilder.java:92]: no MaterializedView.AsyncRefreshContext and RefreshSchemaDes is different one.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 2 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3297]: because createMvSuccess is always true

**Comment by Astralidea** [comment]
> better to add some UT

this case is not easy use UT, already have case to cover.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 56 / 76 (73.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|2|00.00%|[561, 564]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|3|00.00%|[360, 362, 365]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|12|25|48.00%|[320, 528, 529, 530, 531, 532, 533, 534, 535, 537, 3383, 3384, 3385]|
|:large_blue_circle:|com/starrocks/alter/Alter.java|2|3|66.67%|[270]|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|40|41|97.56%|[79]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|2|100.00%|[]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 2 above)
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3291]: no, when there is if not exist, the createMvSuccess may be not true.

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by meegoo — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**